### PR TITLE
add SQLServer grammar create database

### PR DESCRIPTION
**Pull request by Icesource**
For #6478.

Changes proposed in this pull request:

Create SQLServer Grammar create database

Hi, @jingshanglu , I've added SQL definition for SQLServer Create Database statement. Please check it.

### CODE REVIEW

**Review by tristaZero — CHANGES_REQUESTED** [review]
@Icesource Hi thanks for your PR, @jingshanglu could you have a review?

**Comment by Icesource** [comment]
> Hi, @jingshanglu , I've added SQL definition for SQLServer Create Database statement. Please check it.

Hi, @jingshanglu , this is my newly submitted pull request. Could you check it for me ?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/DDLStatement.g4 [hunk 1]
```diff
@@ -450,3 +454,50 @@ onHistoryTableClause
...
+    : FILEGROUP ignoredIdentifier databaseFileGroupContains? fileSpec (COMMA_ fileSpec)*
+    ;
+
+databaseFileGroupContains
+    : (CONTAINS FILESTREAM)? DEFAULT? | CONTAINS MEMORY_OPTIMIZED_DATA
+    ;
+
+databaseLogOns
+    : (LOG ON fileSpec (COMMA_ fileSpec)*)?
+    ;
```
Comment (jingshanglu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/DDLStatement.g4:503]: @Icesource There should be a empty line at the end of file.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/SQLServerKeyword.g4 [hunk 2]
```diff
@@ -1070,3 +1066,72 @@ FILTER_COLUMN
...
+    ;
+
+DB_CHAINING
+    : D B UL_ C H A I N I N G
+    ;
+
+TRUSTWORTHY
+    : T R U S T W O R T H Y
+    ;
+
```
Comment (jingshanglu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/SQLServerKeyword.g4:1137]: @Icesource  If these tokens is not reserved keyword, maybe these should be contained in `unreservedWord `(BaseRule.g4).

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/DDLStatement.g4 (hunk 1 above)
Comment (Icesource) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/DDLStatement.g4]: done

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/SQLServerKeyword.g4 (hunk 2 above)
Comment (Icesource) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/SQLServerKeyword.g4]: @jingshanglu done

**Comment by tristaZero** [comment]
@Icesource Thanks for update, please review one more time @jingshanglu

**Review by jingshanglu — APPROVED** [review]

**Comment by jingshanglu** [comment]
> @Icesource Thanks for update, please review one more time @jingshanglu

@tristaZero you request `request changes`,need you to deal with it.

**Comment by tristaZero** [comment]
Hi @Icesource @jingshanglu When I am going to merge this one, some of the conflicts came out, could you fix them so that I can merge it?

**Comment by Icesource** [comment]
> Hi @Icesource @jingshanglu When I am going to merge this one, some of the conflicts came out, could you fix them so that I can merge it?

@tristaZero No. I will fix the conficts as soon as possible. Please review and merge later. Thank you!

**Comment by Icesource** [comment]
@tristaZero Hello, I have fixed the conflicts

**Review by tristaZero — APPROVED** [review]
Hi @Icesource Sorry for my delayed response. Nice. ;-)

**Pull request merged**